smithy-model/src/main/java/software/amazon/smithy/model/Model.java

```diff
@@ -761,6 +762,26 @@ public <T extends Shape> Stream<T> shapes(Class<T> shapeType) {
         return toSet(shapeType).stream();
     }
 
+    /**
+     * Gets a stream of shapes that have mixins.
+     *
+     * @return stream of shapes
+     */
+    public Stream<Shape> shapesWithMixins() {
+        return shapeMap.values().stream()
+            .filter(shape -> !shape.isMemberShape())
+            .filter(shape -> !shape.getMixins().isEmpty());
+    }
+
+    /**
+     * Gets a stream of shapes defined as mixins.
+     *
+     * @return stream of shapes
+     */
+    public Stream<Shape> mixins() {
+        return shapeMap.values().stream().filter(shape -> shape.hasTrait(MixinTrait.class));
```

Comment:
I don't think this needs a dedicated method, it's `getShapesWithTrait(MixinTrait.class).stream()`.

Reply:
Resolved in commit: 40775d9